scripts/openmc-update-mgxs

```diff
@@ -0,0 +1,225 @@
+
+        # Either add the data to a previously existing xsdata (if it is
+        # for the same 'name' but a different temperature), or create a
+        # new one.
+        try:
+            # It is in our list, so store that entry
+            i = names.index(name)
+        except ValueError:
+            # It is not in our list, so add it
+            i = -1
+            xsd.append(openmc.XSdata(name, energy_groups,
+                                     temperatures=temperatures,
+                                     representation=representation))
+            if awr is not None:
+                xsd[-1].atomic_weight_ratio = awr
+            if representation == 'angle':
+                xsd[-1].num_azimuthal = n_azi
+                xsd[-1].num_polar = n_pol
+            xsd[-1].scatter_format = scatter_format
+            xsd[-1].order = order
+            names.append(name)
+
+        if scatter_format == 'legendre':
+            order_dim = order + 1
+        else:
+            order_dim = order
+
+        if i != -1:
+            xsd[i].add_temperature(temperature)
+
+        temp = get_data(xsdata_elem, 'total')
+        if temp is not None:
+            temp = np.array(temp.split(), dtype=float)
+            total = temp.astype(np.float)
+            total.shape = xsd[i].xs_shapes['[G]']
+            xsd[i].set_total(total, temperature)
+
+        if inverse_velocity is not None:
+            xsd[i].set_inverse_velocity(inverse_velocity, temperature)
+
+        temp = get_data(xsdata_elem, 'absorption')
+        temp = np.array(temp.split())
+        absorption = temp.astype(np.float)
+        absorption.shape = xsd[i].xs_shapes['[G]']
+        xsd[i].set_absorption(absorption, temperature)
+
+        temp = get_data(xsdata_elem, 'scatter')
+        temp = np.array(temp.split())
+        scatter = temp.astype(np.float)
+        # This is now a flattened-array of something that started with a
+        # shape of [Order][G][G']; we need to unflatten and then switch the
+        # ordering
+        in_shape = (order_dim, energy_groups.num_groups,
+                    energy_groups.num_groups)
+        if representation == 'angle':
+            in_shape = (n_pol, n_azi) + in_shape
+            scatter.shape = in_shape
+            scatter = np.swapaxes(scatter, 2, 3)
+            scatter = np.swapaxes(scatter, 3, 4)
+        else:
+            scatter.shape = in_shape
+            scatter = np.swapaxes(scatter, 0, 1)
+            scatter = np.swapaxes(scatter, 1, 2)
+
+        xsd[i].set_scatter_matrix(scatter, temperature)
+
+        temp = get_data(xsdata_elem, 'multiplicity')
+        if temp is not None:
+            temp = np.array(temp.split())
+            multiplicity = temp.astype(np.float)
+            multiplicity.shape = xsd[i].xs_shapes["[G][G']"]
+            xsd[i].set_multiplicity_matrix(multiplicity, temperature)
+
+        temp = get_data(xsdata_elem, 'fission')
```

Comment:
There is an excessive reuse of the same variable `temp`. This could lead to bugs in future if you decide to move pieces of the code around. Consider having different variables for each section. Better yet, separate them in individual methods.